lib/problems.rb

```diff
@@ -1,13 +1,61 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
+
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  opening_braces = ['{','[','(']
+  stack = Stack.new
+  return true if string == ''
+  i = 0
+  while i < string.length
+      if opening_braces.include?(string[i])
+        stack.push(string[i])
+      else
+        current = stack.pop
+        match = false
+        if current == "{" && string[i] == "}"
```

Comment:
Just a note, instead of these if-else blocks, you could have a hash with the key being the close brace and the value being the matching open brace.  Then you can do this with 1 if statement.